code/mission/missioncampaign.cpp

```diff
@@ -1914,6 +1914,55 @@ void mission_campaign_jump_to_mission(char *name)
 	}
 }
 
+/** DahBlount
+* This allows the player to return to the hub mission if a campaign has a campaign map
+*/
+void mission_campaign_return_to_hub(char *name)
+{
+	int i = 0, mission_num = -1;
+	char dest_name[64], *p;
+
+	// load in the campaign junk
+	mission_load_up_campaign();
+
+	// tack the .fs2 onto the input name
+	strcpy_s(dest_name, name);
+	p = strchr(dest_name, '.');
+	if (p != NULL)
+		*p = '\0';
+	strcat_s(dest_name, ".fs2");
+
+	// search for our mission
+	for (i = 0; i < Campaign.num_missions; i++) {
+		if ((Campaign.missions[i].name != NULL) && !stricmp(Campaign.missions[i].name, dest_name)) {
+			mission_num = i;
+			break;
+		}
+	}
+
+	if (mission_num < 0) {
+		// if we got here, no match was found
+		// restart the campaign
+		mission_campaign_savefile_delete(Campaign.filename);
+		mission_campaign_load(Campaign.filename);
+	}
+	else {
+		for (i = 0; i < MAX_SHIP_CLASSES; i++) {
```

Comment:
It might actually be good to switch to `Ship_info.size()` now that I think about it, having another opinion would be great.

Same thing with `Num_weapon_types`.